src/app/backend/resource/event/common.go

```diff
@@ -169,6 +171,18 @@ func FillEventsType(events []v1.Event) []v1.Event {
 
 // ToEvent converts event api Event to Event model object.
 func ToEvent(event v1.Event) common.Event {
+	firstTimestamp, lastTimestamp := event.FirstTimestamp, event.LastTimestamp
+
+	if !event.EventTime.IsZero() {
+		eventTime := metaV1.NewTime(time.Unix(event.EventTime.Unix(), 0))
+		if event.FirstTimestamp.IsZero() {
+			firstTimestamp = eventTime
+		}
+		if event.LastTimestamp.IsZero() {
+			lastTimestamp = firstTimestamp
+		}
+	}
+
```

Comment:
I'd maybe simplify this code to
```go
	firstTimestamp, lastTimestamp := event.FirstTimestamp, event.LastTimestamp
	eventTime := metaV1.NewTime(event.EventTime.Time)

	if firstTimestamp.IsZero() {
		firstTimestamp = eventTime
	}

	if lastTimestamp.IsZero() {
		lastTimestamp = eventTime
	}
```

If it's already zero then overriding with zero does not matter for us. I know it adds a couple of extra cycles, but it's very minor and the code is simpler.

Reply:
Yes, this makes the code more readable. I have updated.